docs/src/configuration.md

```diff
@@ -108,31 +31,30 @@ The following MPI implementations should work out-of-the-box with MPI.jl:
 - [Fujitsu MPI](https://www.fujitsu.com/global/about/resources/publications/technicalreview/2020-03/article07.html#cap-03)
 - [HPE MPT/HMPT](https://support.hpe.com/hpesc/public/docDisplay?docLocale=en_US&docId=a00105727en_us)
 
-If the implementation is changed, you will need to use [`MPI.use_system_binary()`](@ref MPIPreferences.use_system_binary)
-or [`MPI.use_jll_binary()`](@ref MPIPreferences.use_jll_binary).
+### Configuration
 
-#### Advanced options
+Run `MPIPreferences.use_system_binary()`. This will attempt to locate and to identify any available MPI implementation, and create a file called `LocalPreferences.toml` adjacent to the current `Project.toml`.
 
-```@doc
-MPI.use_system_binary
+```sh
+julia --project -e 'using MPI; MPIPreferences.use_system_binary()'
 ```
 
-You can use the argument `mpiexec` to provide the name (or full path) of the MPI launcher executable. The default is
-`mpiexec`, but some clusters require using the scheduler launcher interface (e.g. `srun`
-on Slurm, `aprun` on PBS). If the MPI library has an uncommon name you can provide it in `library_names`.
-The MPI standard does not specify the exact application binary interface (ABI).
-In case ABI detection fails you can provide a manual choice (either `MPICH`, `MPItrampoline`, `OpenMPI`, or `MicrosoftMPI`),
-but also open an issue such that the automatic detection can be improved.
-`export_prefs=true` can be used to copy the preferences into the `Project.toml` instead of creating a
-`LocalPreferences.toml` file to hold them.
+If the implementation is changed, you will need to call this function again. See the [`MPIPreferences.use_system_binary`](@ref) documentation for specific options.
 
-#### Notes to HPC cluster adminstators
+!!! note
+    You can copy `LocalPreferences.toml` to a different project folder, but you must list
+    `MPIPreferences` in the `[extras]` section of the `Project.toml` for the settings
+    to take effect. Due to a bug in Julia (until `v1.6.5` and `v1.7.1`), getting preferences
+    from transitive dependencies is broken (https://github.com/JuliaPackaging/Preferences.jl/issues/24).
+    To fix this update your version of Julia, or add `MPIPreferences` as a direct dependency to your project.
```

Comment:
This note was here also before, but I suspect as it's phrased now it may be a bit confusing.  As far as I understand, `MPIPreferences` should be in the active environment, which *for running tests* means that it should be in `[extras]`, right?  This sounds like it should *always* be in `[extras]`.  Or am I getting it wrong?

Reply:
I think it also works if it is in the `[deps]` section?

Reply:
I've updated this, let me know what you think.

Reply:
Yes, I think it's better now, thanks!